jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp

```diff
@@ -169,7 +169,50 @@ static ecma_completion_value_t
 ecma_builtin_string_prototype_object_char_code_at (ecma_value_t this_arg, /**< this argument */
                                                    ecma_value_t arg) /**< routine's argument */
 {
-  ECMA_BUILTIN_CP_UNIMPLEMENTED (this_arg, arg);
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+  /* 1 */
+  ECMA_TRY_CATCH (check_coercible_val,
+                  ecma_op_check_object_coercible (this_arg),
+                  ret_value);
+
+  /* 2 */
+  ECMA_TRY_CATCH (to_string_val,
+                  ecma_op_to_string (this_arg),
+                  ret_value);
+
+  /* 3 */
+  ECMA_OP_TO_NUMBER_TRY_CATCH (index_num,
+                               arg,
+                               ret_value);
+
+  /* 4 */
+  ecma_string_t *original_string_p = ecma_get_string_from_value (to_string_val);
+  const ecma_length_t len = ecma_string_get_length (original_string_p);
+
+  ecma_number_t *ret_num_p = ecma_alloc_number ();
+
+  /* 5 */
+  if (index_num < 0 || index_num >= len || !len)
+  {
+    *ret_num_p = ecma_number_make_nan ();
+  }
+  else
+  {
+    /* 6 */
+    ecma_char_t new_ecma_char = ecma_string_get_char_at_pos (original_string_p, ecma_number_to_uint32 (index_num));
```

Comment:
While actual length of strings, currently, can't exceed `uint32_t` range, value of `index_num` can be bigger. `ToInteger` is not equivalent to `ToUInt32`, as `ToUInt32` performs "modulo 2^32" operation on its argument, and `ToInteger` performs `floor`.

The code would work correctly, as `index_num` doesn't exceed `len`, as checked above.

Could you, please, add an assertion that `ecma_number_to_uint32 (index_num) == ecma_number_trunc (index_num)` and add comment, describing why `ToInteger` and `ToUInt32` are equivalent in the case?